src/connection/database/actions.js

```diff
@@ -63,11 +63,16 @@ export function signUp(id) {
 
     if (!additionalSignUpFields(m).isEmpty()) {
       params.user_metadata = {};
+      params.user_metadata.extraFields = {};
       additionalSignUpFields(m).forEach(x => {
-        params.user_metadata[x.get('name')] = c.getFieldValue(m, x.get('name'));
+        if (x.get('isExtra')) {
```

Comment:
extra fields go into its own object instead of directly to the user_metadata object